src/languageservice/services/yamlSchemaService.ts

```diff
@@ -488,13 +492,59 @@ export class YAMLSchemaService extends JSONSchemaService {
     }
   }
 
+  /*
+   * Everything below here is needed because we're importing from vscode-json-languageservice umd and we need
+   * to provide a wrapper around the javascript methods we are calling since they have no type
+   */
+
   getOrAddSchemaHandle(id: string, unresolvedSchemaContent?: JSONSchema): SchemaHandle {
     return super.getOrAddSchemaHandle(id, unresolvedSchemaContent);
   }
 
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  loadSchema(schemaUri: string): Thenable<any> {
-    return super.loadSchema(schemaUri);
+  loadSchema(schemaUri: string): Thenable<UnresolvedSchema> {
+    const requestService = this.requestService;
+    return super.loadSchema(schemaUri).then((unresolvedJsonSchema: UnresolvedSchema) => {
+      // If json-language-server failed to parse the schema, attempt to parse it as YAML instead.
+      if (unresolvedJsonSchema.errors && unresolvedJsonSchema.schema === undefined) {
+        return requestService(schemaUri).then(
+          (content) => {
+            if (!content) {
+              const errorMessage = localize(
+                'json.schema.nocontent',
+                "Unable to load schema from '{0}': No content.",
+                toDisplayString(schemaUri)
+              );
+              return new UnresolvedSchema(<JSONSchema>{}, [errorMessage]);
+            }
+
+            try {
+              const schemaContent: JSONSchema = yaml.safeLoad(content);
```

Comment:
If you are doing a simple conversion then I'm concerned that it might not work with k8s yaml schemas. The structure of which has the schema held within it.

Reply:
@joshuawilson I'm not sure that understand your point, can you add sample of such k8s yaml scheme?

Reply:
Here is a link to the Knative Eventing Subscription schema. Notice that is has structure wrapping the actual schema. 
https://github.com/knative/eventing/blob/master/config/core/resources/subscription.yaml

Reply:
No, because that file itself is not YAML Scheme, it is k8's CRD. To get actual scheme you need to extract that [yaml](https://github.com/knative/eventing/blob/master/config/core/resources/subscription.yaml#L31-L277) in to separate yaml file

Reply:
This is a fine, I was just hoping to be able to point to the URL for a k8s schema and use it. But I do understand that has become a more generic tool.